azkaban-exec-server/src/main/java/azkaban/execapp/AzkabanExecutorServer.java

```diff
@@ -551,12 +559,29 @@ public Object getMBeanAttribute(ObjectName name, String attribute) {
    *
    * @return hostname
    */
-  public String getHost() {
-    String host = "unkownHost";
+  public static String getHost() {
+    if(StringUtils.isEmpty(hostAlias)) {
+      URL scriptUrl = Resources.getResource(ServerInternals.HOST_NAME_ALIAS_SCRIPT);
+      if (scriptUrl != null) {
+        try {
+          File aliasFile = new File(scriptUrl.toURI());
+          hostAlias = FileIOUtils.runShellCommand("perl " + aliasFile, new File(".")).trim();
+        } catch (Exception e) {
+          logger.error("Failed to get alias", e);
+        }
+      }
+      if(StringUtils.isEmpty(hostAlias) || scriptUrl == null) {
+        hostAlias = "unknownAlias";
+      }
+    }
+
```

Comment:
Should we check the obtained URL a valid/legal VIP somewhere?  It might be helpful if perl gives a wrong vip name.